src/actions/ensure-package-json.ts

```diff
@@ -0,0 +1,51 @@
+import * as nodeFS from 'node:fs';
+import ora from 'ora';
+import { shell } from '../shell';
+import type { Context } from './context';
+
+function hasPackageJson(dir: string): boolean {
+	try {
+		const normalizedDir = dir.replace(/^\/+/, '');
+
+		if (!nodeFS.existsSync(normalizedDir)) {
+			nodeFS.mkdirSync(normalizedDir, { recursive: true });
+		}
+
+		const files = nodeFS.readdirSync(normalizedDir);
+		return files.includes('package.json');
+	} catch (error: unknown) {
+		return false;
+	}
+}
+
+async function initializePackageJson(context: Context): Promise<void> {
+	const spinner = ora('Creating a `package.json` file...').start();
+
+	if (context.dryRun) {
+		spinner.info('Creating a `package.json` file... (skipped due to --dry-run)');
+		spinner.stop();
+		return;
+	}
+
+	try {
+		await shell(context.packageManager, ['init', '-y'], { cwd: context.cwd });
```

Comment:
question

Should we create the file ourselves instead?
I am asking because I would expect for us to cleanup the other files that gets created by the different package managers.
e.g. [`bun` adds 4 more files](https://bun.sh/docs/cli/init)

Reply:
Considering that the command to install a package creates `package.json` - maybe creating it on our side is really unnecessary. I'll think about later.